modules/core/04-channel/keeper/keeper.go

```diff
@@ -541,6 +547,12 @@ func (k Keeper) deleteUpgrade(ctx sdk.Context, portID, channelID string) {
 	store.Delete(host.ChannelUpgradeKey(portID, channelID))
 }
 
+// hasCounterpartyUpgrade returns true if a counterparty upgrade exists in store
+func (k Keeper) hasCounterpartyUpgrade(ctx sdk.Context, portID, channelID string) bool {
```

Comment:
copied over same func @AdityaSripal used for `HasUpgrade` but chose to leave the `hasCounterpartyUpgrade` private since I don't see a reason to add it to the public API when it is used only for defensive checks in edge case scenarios